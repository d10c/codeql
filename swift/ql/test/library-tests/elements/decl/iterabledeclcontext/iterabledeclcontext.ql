import swift

from IterableDeclContext decl, int index, Decl member
where
  decl.(Decl).getModule().getName() = "iterabledeclcontext" and
  member = decl.getMember(index)
select decl, index, member, member.getPrimaryQlClasses()
