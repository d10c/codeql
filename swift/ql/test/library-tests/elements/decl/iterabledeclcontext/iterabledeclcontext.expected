| Class1 | 0 | iterabledeclcontext.swift:13:2:13:10 | init() | ConstructorDecl |
| Class1 | 1 | iterabledeclcontext.swift:14:2:14:10 | deinit() | DestructorDecl |
| Class1 | 2 | iterabledeclcontext.swift:16:2:16:27 | var ... = ... | PatternBindingDecl |
| Class1 | 3 | iterabledeclcontext.swift:16:6:16:6 | constantField | ConcreteVarDecl |
| Class1 | 4 | iterabledeclcontext.swift:16:6:16:6 | get | AccessorDecl |
| Class1 | 5 | iterabledeclcontext.swift:17:2:17:27 | var ... = ... | PatternBindingDecl |
| Class1 | 6 | iterabledeclcontext.swift:17:6:17:6 | mutableField | ConcreteVarDecl |
| Class1 | 7 | iterabledeclcontext.swift:17:6:17:6 | get | AccessorDecl |
| Class1 | 8 | iterabledeclcontext.swift:17:6:17:6 | set | AccessorDecl |
| Class1 | 9 | iterabledeclcontext.swift:17:6:17:6 | (unnamed function decl) | AccessorDecl |
| Class1 | 10 | iterabledeclcontext.swift:18:2:18:40 | var ... = ... | PatternBindingDecl |
| Class1 | 11 | iterabledeclcontext.swift:18:13:18:13 | staticConstantField | ConcreteVarDecl |
| Class1 | 12 | iterabledeclcontext.swift:18:13:18:13 | get | AccessorDecl |
| Class1 | 13 | iterabledeclcontext.swift:19:2:19:40 | var ... = ... | PatternBindingDecl |
| Class1 | 14 | iterabledeclcontext.swift:19:13:19:13 | staticMutableField | ConcreteVarDecl |
| Class1 | 15 | iterabledeclcontext.swift:19:13:19:13 | get | AccessorDecl |
| Class1 | 16 | iterabledeclcontext.swift:19:13:19:13 | set | AccessorDecl |
| Class1 | 17 | iterabledeclcontext.swift:19:13:19:13 | (unnamed function decl) | AccessorDecl |
| Class1 | 18 | iterabledeclcontext.swift:21:2:23:2 | var ... = ... | PatternBindingDecl |
| Class1 | 19 | iterabledeclcontext.swift:21:12:21:12 | gettableClassField | ConcreteVarDecl |
| Class1 | 20 | iterabledeclcontext.swift:21:36:23:2 | get | AccessorDecl |
| Class1 | 21 | iterabledeclcontext.swift:24:2:27:2 | var ... = ... | PatternBindingDecl |
| Class1 | 22 | iterabledeclcontext.swift:24:6:24:6 | observableField | ConcreteVarDecl |
| Class1 | 23 | iterabledeclcontext.swift:25:3:25:36 | willSet | AccessorDecl |
| Class1 | 24 | iterabledeclcontext.swift:26:3:26:34 | didSet | AccessorDecl |
| Class1 | 25 | iterabledeclcontext.swift:24:6:24:6 | get | AccessorDecl |
| Class1 | 26 | iterabledeclcontext.swift:24:6:24:6 | set | AccessorDecl |
| Class1 | 27 | iterabledeclcontext.swift:24:6:24:6 | (unnamed function decl) | AccessorDecl |
| Class1 | 28 | iterabledeclcontext.swift:29:2:36:2 | subscript ... | SubscriptDecl |
| Class1 | 29 | iterabledeclcontext.swift:30:3:32:3 | get | AccessorDecl |
| Class1 | 30 | iterabledeclcontext.swift:33:3:35:3 | set | AccessorDecl |
| Class1 | 31 | iterabledeclcontext.swift:29:2:29:2 | (unnamed function decl) | AccessorDecl |
| Class1 | 32 | iterabledeclcontext.swift:38:2:38:25 | instanceMethod() | ConcreteFuncDecl |
| Class1 | 33 | iterabledeclcontext.swift:39:2:39:30 | staticMethod() | ConcreteFuncDecl |
| Class1 | 34 | iterabledeclcontext.swift:40:2:40:28 | classMethod() | ConcreteFuncDecl |
| Class1 | 35 | iterabledeclcontext.swift:44:2:44:25 | Associated | TypeAliasDecl |
| Class1 | 36 | iterabledeclcontext.swift:48:2:50:2 | NestedClass | ClassDecl |
| Class1 | 37 | iterabledeclcontext.swift:51:2:53:2 | NestedStruct | StructDecl |
| Class1 | 38 | iterabledeclcontext.swift:54:2:58:2 | NestedEnum | EnumDecl |
| NestedClass | 0 | iterabledeclcontext.swift:49:3:49:25 | foo() | ConcreteFuncDecl |
| NestedClass | 1 | file://:0:0:0:0 | Associated | TypeAliasDecl |
| NestedClass | 2 | iterabledeclcontext.swift:48:8:48:8 | deinit() | DestructorDecl |
| NestedClass | 3 | iterabledeclcontext.swift:48:8:48:8 | init() | ConstructorDecl |
| NestedEnum | 0 | iterabledeclcontext.swift:55:3:55:8 | case ... | EnumCaseDecl |
| NestedEnum | 1 | iterabledeclcontext.swift:55:8:55:8 | case1 | EnumElementDecl |
| NestedEnum | 2 | iterabledeclcontext.swift:56:3:56:8 | case ... | EnumCaseDecl |
| NestedEnum | 3 | iterabledeclcontext.swift:56:8:56:8 | case2 | EnumElementDecl |
| NestedEnum | 4 | iterabledeclcontext.swift:57:3:57:25 | foo() | ConcreteFuncDecl |
| NestedEnum | 5 | file://:0:0:0:0 | __derived_enum_equals(_:_:) | ConcreteFuncDecl |
| NestedEnum | 6 | file://:0:0:0:0 | var ... = ... | PatternBindingDecl |
| NestedEnum | 7 | file://:0:0:0:0 | hash(into:) | ConcreteFuncDecl |
| NestedEnum | 8 | file://:0:0:0:0 | Associated | TypeAliasDecl |
| NestedEnum | 9 | file://:0:0:0:0 | hashValue | ConcreteVarDecl |
| NestedEnum | 10 | file://:0:0:0:0 | get | AccessorDecl |
| NestedStruct | 0 | iterabledeclcontext.swift:52:3:52:25 | foo() | ConcreteFuncDecl |
| NestedStruct | 1 | file://:0:0:0:0 | Associated | TypeAliasDecl |
| NestedStruct | 2 | iterabledeclcontext.swift:51:9:51:9 | init() | ConstructorDecl |
| Protocol1 | 0 | iterabledeclcontext.swift:5:2:5:30 | Associated | AssociatedTypeDecl |
| Protocol1 | 1 | iterabledeclcontext.swift:6:2:6:16 | foo() | ConcreteFuncDecl |
| extension | 0 | iterabledeclcontext.swift:9:2:9:45 | foo() | ConcreteFuncDecl |
